bitarray/_bitarray.c

```diff
@@ -2068,38 +2051,96 @@ bitarray_cpinvert(bitarrayobject *self)
     return res;
 }
 
-#define BITWISE_FUNC(oper)  \
+#if HAS_VECTORS
+#define BITWISE_FUNC_INTERNAL(SELF, OTHER, OP, OPEQ) do {         \
+    Py_ssize_t i = 0;                                             \
+                                                                  \
+    for (; i + sizeof(vec) < Py_SIZE(SELF); i += sizeof(vec)) {   \
```

Comment:
If you cache size in a local variable, compiler will have an easier time proving that it doesn't change inside the loop.